processor/elasticprocessor/internal/hostmetrics/disk.go

```diff
@@ -0,0 +1,152 @@
+package hostmetrics
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+func addDiskMetrics(metrics pmetric.MetricSlice, resource pcommon.Resource, dataset string) error {
+	for i := 0; i < metrics.Len(); i++ {
+		metric := metrics.At(i)
+		if metric.Name() == "system.disk.io" || metric.Name() == "system.disk.operations" {
+			dataPoints := metric.Sum().DataPoints()
+			for j := 0; j < dataPoints.Len(); j++ {
+				dp := dataPoints.At(j)
+
+				var device string
+				if d, ok := dp.Attributes().Get("device"); ok {
+					device = d.Str()
+				} else {
+					continue
+				}
+
+				if direction, ok := dp.Attributes().Get("direction"); ok {
+					name := metric.Name()
+					timestamp := dp.Timestamp()
+					value := dp.IntValue()
+
+					switch direction.Str() {
```

Comment:
L30-34 can be simplified to just `addDiskIntMetric(metrics, resource, dataset, name, device, direction.Str(), timestamp, value)`

Reply:
Done!